client/src/leap/soledad/client/encdecpool.py

```diff
@@ -212,29 +174,7 @@ def enqueue_doc_for_encryption(self, doc):
         :param doc: The document to be encrypted.
         :type doc: SoledadDocument
         """
-        try:
-            self._encr_queue.put(doc)
-        except Queue.Full:
-            # do not asynchronously encrypt this file if the queue is full
-            pass
-
-    @defer.inlineCallbacks
-    def _maybe_encrypt_and_recurse(self):
-        """
-        Process one document from the encryption queue.
-
-        Asynchronously encrypt a document that will then be stored in the sync
-        db. Processed documents will be read by the SoledadSyncTarget during
-        the sync_exchange.
-        """
-        try:
-            while self.running:
-                doc = yield self._encr_queue.get()
-                self._encrypt_doc(doc)
-        except defer.QueueUnderflow:
-            self._delayed_call = reactor.callLater(
-                self.ENCRYPT_LOOP_PERIOD,
-                self._maybe_encrypt_and_recurse)
+        self._encrypt_doc(doc)
 
     def _encrypt_doc(self, doc):
```

Comment:
As this is not used anywhere besides `enqueue_doc_for_encryption` and this method consist only in a call to `_encrypt_doc` I would move the code of `_encrypt_doc` there.